feature: adding support custom username field

InzGIBA opened the pull request:
I faced a problem when implementing this package in my project. The administrator panel worked perfectly when authorizing, but for example package simplejwt (drf) sent data based on a custom field of the user.

Link: https://github.com/jazzband/djangorestframework-simplejwt/blob/c791e987332ed5e22a86428160d6372b1d85ffae/rest_framework_simplejwt/serializers.py#L41

In this pull-request we added a check for the "username" field

Link to Django ModelBackend: https://github.com/django/django/blob/ae10146793dca0e0594c7acdee20ca3810983f39/django/contrib/auth/backends.py#L38

francoisfreitag commented:
The suggestion looks fine to me. Could you please add a test to cover the added code path?

InzGIBA commented:
- Fixed a bug that prevented all previously written tests from being run in projects
- Added a test "test_auth_custom_field" to test a different argument in the "authenticate" function, in turn the test was based on an example of another test "test_get_custom_field"

francoisfreitag reviewed (approved):
Looking great, thanks!
